### PR TITLE
Remove gateway datatypes

### PR DESCRIPTION
**Pull request by nielsm5**

### CODE REVIEW

File: management-gateway/src/main/java/org/frankframework/management/bus/OutboundGateway.java [hunk 1]
```diff
@@ -16,20 +16,26 @@
...
-public interface OutboundGateway<T> extends IntegrationPattern {
+public interface OutboundGateway extends IntegrationPattern {
+
+	@Override
+	default IntegrationPatternType getIntegrationPatternType() {
+		return IntegrationPatternType.outbound_gateway;
+	}
 
 	/**
 	 * T in T out.
```
Comment (jkosternl) [inline management-gateway/src/main/java/org/frankframework/management/bus/OutboundGateway.java:30]:
```suggestion
	 * I in O out.
```

File: management-gateway/src/main/java/org/frankframework/management/bus/OutboundGateway.java [hunk 2]
```diff
@@ -16,20 +16,26 @@
...
 	 * @param in Message to send
 	 * @return Response message
 	 */
-	public Message<T> sendSyncMessage(Message<T> in);
+	public <I, O> Message<O> sendSyncMessage(Message<I> in);
 
 	/**
 	 * T in, no reply
```
Comment (jkosternl) [inline management-gateway/src/main/java/org/frankframework/management/bus/OutboundGateway.java:37]:
```suggestion
	 * I in, no reply
```

File: checkstyle.xml [hunk 3]
```diff
@@ -54,7 +54,7 @@
 		<module name="IllegalImport">
 			<property name="regexp" value="true" />
 			<!-- Reject any org.junit import that's not also org.junit.jupiter: -->
-			<property name="illegalClasses" value="^org\.junit\.(?!jupiter\.).+" />
+			<property name="illegalClasses" value="^org\.junit\.(?!(jupiter)|(platform)\.).+" />
```
Comment (jkosternl) [inline checkstyle.xml:57]: Lovely regex 😉

**Review by jkosternl — APPROVED** [review]

File: checkstyle.xml (hunk 3 above)
Comment (evandongen) [reply checkstyle.xml:57]: Maybe update the comment above as well?

File: checkstyle.xml [hunk 4]
```diff
@@ -54,7 +54,7 @@
 		<module name="IllegalImport">
 			<property name="regexp" value="true" />
 			<!-- Reject any org.junit import that's not also org.junit.jupiter: -->
```
Comment (nielsm5) [inline checkstyle.xml:56]:
```suggestion
			<!-- Reject any org.junit import that's not also org.junit.jupiter or org.junit.platform (junit extensions) -->
```

**Pull request merged**